frontend-thea-aa/src/routes/+page.svelte

```diff
@@ -4,42 +4,176 @@
-
+	import { Building, Building2, Search } from 'lucide-svelte';
+	import { Cloudy } from 'lucide-svelte';
+	import { Wind } from 'lucide-svelte';
+	import { ListFilter } from 'lucide-svelte';
+	import AirQualityRanking from '$lib/components/AirQualityRanking.svelte'
+   
 	let { data } = $props()
+	let currentView = $state('');
+	let height = $state(0);
+	let mapInstance: any;
+	let apiKey = "672fc9cf41928442178811nqtede4fc"
 
-	let showContaminated = $state(true)
-	let showEmissionEvents = $state(true)
-</script>
-
-<main class="flex">
-	<aside>
-		<input placeholder="search" />
-		<div>
-			<Button
-				on:click={() => {
-					showContaminated = !showContaminated
-				}}>Contaminated Sites</Button
-			>
-			<Button
-				on:click={() => {
-					showEmissionEvents = !showEmissionEvents
-				}}>Emission Events</Button
-			>
-		</div>
+	let searchAddress = $state('');
+	let searchRadius = $state('');
+	let isLoading = $state(false);
+	let filteredData = $state({
+		contaminatedSites: data.contaminatedSites,
+		emissionEvents: data.emissionEvents,
+		airQualitySites: data.airQualitySites
+	});
 
-		<!-- <AirQualityRanking items={data.airQualitySites} /> -->
-		{#if showContaminated}
-			<ContaminatedSites items={data.contaminatedSites} />
-		{/if}
+	function calculateDistance(lat1: number, long1: number, lat2: number, long2: number): number {
+		const R = 3959; // Earth's radius in miles
+		const dLat = (lat2 - lat1) * Math.PI / 180;
+		const dLon = (long2 - long1) * Math.PI / 180;
+		const a = 
+			Math.sin(dLat/2) * Math.sin(dLat/2) +
+			Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) * 
+			Math.sin(dLon/2) * Math.sin(dLon/2);
+		const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
+		return R * c;
+	}
+
+	async function geocodeAddress(address: string): Promise<{lat: number, long: number} | null> {
+		try {
+			const response = await fetch(
+				`https://geocode.maps.co/search?q=${encodeURIComponent(address)}&api_key=${encodeURIComponent(apiKey)}`
+			);
+			
+			if (!response.ok) {
+				throw new Error(`Geocoding failed with status: ${response.status}`);
+			}
+
+			const data = await response.json();
+			
+			if (data && data.length > 0) {
+				return {
+					lat: parseFloat(data[0].lat),
+					long: parseFloat(data[0].lon)
+				};
+			}
+			return null;
+		} catch (error) {
+			console.error('Geocoding error:', error);
+			return null;
+		}
+	}
 
-		{#if showEmissionEvents}
-			<EmissionEvents items={data.emissionEvents} />
+	async function applyFilter() {
+		if (!searchAddress || !searchRadius) {
+			// Reset to original data if no filter criteria
+			filteredData = {
+				contaminatedSites: data.contaminatedSites,
+				emissionEvents: data.emissionEvents,
+				airQualitySites: data.airQualitySites
+			};
+			return;
+		}
+
+		isLoading = true;
+
+		try {
+			const addressCoords = await geocodeAddress(searchAddress);
+			if (!addressCoords) {
+				alert('Could not find the specified address');
```

Comment:
alert is not ideal, ideally replace with some sort of error in the ui, maybe a toast or something